### DISCUSSION TITLE
How to connect over a Postgres socket?

### DISCUSSION THREAD

**Discussion by charmoniumQ** [firefly-iii/firefly-iii#8395] [Firefly III]
This is my `.env`:

```
    DB_CONNECTION = "pgsql";
    DB_SOCKET = "/run/postgresql";
    DB_DATABASE = cfg.dbname;
    DB_USERNAME = cfg.user;
```

where `php artisan migrate` results in the following error (press expand to see the full error):

<details>
<summary>
<pre><code>
   Illuminate\Database\QueryException
  SQLSTATE[08006] [7] invalid integer value "sslmode=prefer" for connection option "port" (Connection: pgsql, SQL: select * from information_schema.tables where table_catalog = firefly and table_schema = public and table_name = migrations and table_type = 'BASE TABLE')
</code></pre>
</summary>
<pre><code>
[2024-01-15 11:12:01] local.ERROR: Exception is: {"class":"Illuminate\\Database\\QueryException","errorMessage":"SQLSTATE[08006] [7] invalid integer value \"sslmode=prefer\" for connection option \"port\" (Connection: pgsql, SQL: select * from information_schema.tables where table_catalog = firefly and table_schema = public and table_name = migrations and table_type = 'BASE TABLE')","time":"Mon, 15 Jan 2024 11:12:01 +0000","file":"\/nix\/store\/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii\/vendor\/laravel\/framework\/src\/Illuminate\/Database\/Connection.php","line":801,"code":7,"version":"6.0.30","url":"https:\/\/firefly.samgrayson.me","userAgent":"Symfony","json":true,"method":"GET"}
[2024-01-15 11:12:01] local.ERROR: SQLSTATE[08006] [7] invalid integer value "sslmode=prefer" for connection option "port" (Connection: pgsql, SQL: select * from information_schema.tables where table_catalog = firefly and table_schema = public and table_name = migrations and table_type = 'BASE TABLE') {"exception":"[object] (Illuminate\\Database\\QueryException(code: 7): SQLSTATE[08006] [7] invalid integer value \"sslmode=prefer\" for connection option \"port\" (Connection: pgsql, SQL: select * from information_schema.tables where table_catalog = firefly and table_schema = public and table_name = migrations and table_type = 'BASE TABLE') at /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php:801)
[stacktrace]
#0 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(755): Illuminate\\Database\\Connection->runQueryCallback('select * from i...', Array, Object(Closure))
#1 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(407): Illuminate\\Database\\Connection->run('select * from i...', Array, Object(Closure))
#2 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(394): Illuminate\\Database\\Connection->select('select * from i...', Array, false)
#3 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Schema/PostgresBuilder.php(51): Illuminate\\Database\\Connection->selectFromWriteConnection('select * from i...', Array)
#4 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Migrations/DatabaseMigrationRepository.php(184): Illuminate\\Database\\Schema\\PostgresBuilder->hasTable('migrations')
#5 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Migrations/Migrator.php(700): Illuminate\\Database\\Migrations\\DatabaseMigrationRepository->repositoryExists()
#6 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(140): Illuminate\\Database\\Migrations\\Migrator->repositoryExists()
#7 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Support/helpers.php(248): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->Illuminate\\Database\\Console\\Migrations\\{closure}(1)
#8 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(140): retry(1, Object(Closure), 0, Object(Closure))
#9 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(116): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->repositoryExists()
#10 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(84): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->prepareDatabase()
#11 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Migrations/Migrator.php(633): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->Illuminate\\Database\\Console\\Migrations\\{closure}()
#12 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(83): Illuminate\\Database\\Migrations\\Migrator->usingConnection(NULL, Object(Closure))
#13 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->handle()
#14 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#15 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#16 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#17 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#18 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\\Container\\Container->call(Array)
#19 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/symfony/console/Command/Command.php(326): Illuminate\\Console\\Command->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#20 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Console/Command.php(180): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#21 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/symfony/console/Application.php(1081): Illuminate\\Console\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#22 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/symfony/console/Application.php(320): Symfony\\Component\\Console\\Application->doRunCommand(Object(Illuminate\\Database\\Console\\Migrations\\MigrateCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#23 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/symfony/console/Application.php(174): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#24 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#25 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/artisan(35): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#26 {main}
[previous exception] [object] (PDOException(code: 7): SQLSTATE[08006] [7] invalid integer value \"sslmode=prefer\" for connection option \"port\" at /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php:65)
[stacktrace]
#0 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(65): PDO->__construct('pgsql:host=;dbn...', 'firefly', Object(SensitiveParameterValue), Array)
#1 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connectors/Connector.php(44): Illuminate\\Database\\Connectors\\Connector->createPdoConnection('pgsql:host=;dbn...', 'firefly', '', Array)
#2 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connectors/PostgresConnector.php(35): Illuminate\\Database\\Connectors\\Connector->createConnection('pgsql:host=;dbn...', Array, Array)
#3 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connectors/ConnectionFactory.php(184): Illuminate\\Database\\Connectors\\PostgresConnector->connect(Array)
#4 [internal function]: Illuminate\\Database\\Connectors\\ConnectionFactory->Illuminate\\Database\\Connectors\\{closure}()
#5 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(1292): call_user_func(Object(Closure))
#6 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(521): Illuminate\\Database\\Connection->getPdo()
#7 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(416): Illuminate\\Database\\Connection->getPdoForSelect(false)
#8 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(788): Illuminate\\Database\\Connection->Illuminate\\Database\\{closure}('select * from i...', Array)
#9 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(755): Illuminate\\Database\\Connection->runQueryCallback('select * from i...', Array, Object(Closure))
#10 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(407): Illuminate\\Database\\Connection->run('select * from i...', Array, Object(Closure))
#11 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(394): Illuminate\\Database\\Connection->select('select * from i...', Array, false)
#12 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Schema/PostgresBuilder.php(51): Illuminate\\Database\\Connection->selectFromWriteConnection('select * from i...', Array)
#13 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Migrations/DatabaseMigrationRepository.php(184): Illuminate\\Database\\Schema\\PostgresBuilder->hasTable('migrations')
#14 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Migrations/Migrator.php(700): Illuminate\\Database\\Migrations\\DatabaseMigrationRepository->repositoryExists()
#15 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(140): Illuminate\\Database\\Migrations\\Migrator->repositoryExists()
#16 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Support/helpers.php(248): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->Illuminate\\Database\\Console\\Migrations\\{closure}(1)
#17 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(140): retry(1, Object(Closure), 0, Object(Closure))
#18 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(116): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->repositoryExists()
#19 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(84): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->prepareDatabase()
#20 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Migrations/Migrator.php(633): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->Illuminate\\Database\\Console\\Migrations\\{closure}()
#21 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Console/Migrations/MigrateCommand.php(83): Illuminate\\Database\\Migrations\\Migrator->usingConnection(NULL, Object(Closure))
#22 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(36): Illuminate\\Database\\Console\\Migrations\\MigrateCommand->handle()
#23 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Util.php(41): Illuminate\\Container\\BoundMethod::Illuminate\\Container\\{closure}()
#24 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(93): Illuminate\\Container\\Util::unwrapIfClosure(Object(Closure))
#25 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Container/BoundMethod.php(35): Illuminate\\Container\\BoundMethod::callBoundMethod(Object(Illuminate\\Foundation\\Application), Array, Object(Closure))
#26 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(662): Illuminate\\Container\\BoundMethod::call(Object(Illuminate\\Foundation\\Application), Array, Array, NULL)
#27 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Console/Command.php(211): Illuminate\\Container\\Container->call(Array)
#28 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/symfony/console/Command/Command.php(326): Illuminate\\Console\\Command->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#29 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Console/Command.php(180): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Illuminate\\Console\\OutputStyle))
#30 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/symfony/console/Application.php(1081): Illuminate\\Console\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#31 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/symfony/console/Application.php(320): Symfony\\Component\\Console\\Application->doRunCommand(Object(Illuminate\\Database\\Console\\Migrations\\MigrateCommand), Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#32 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/symfony/console/Application.php(174): Symfony\\Component\\Console\\Application->doRun(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#33 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(201): Symfony\\Component\\Console\\Application->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#34 /nix/store/pzcbk7wwqvi53dn4limw52m0qkzrg4s8-firefly-iii/artisan(35): Illuminate\\Foundation\\Console\\Kernel->handle(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\ConsoleOutput))
#35 {main}
"}
   Illuminate\Database\QueryException
  SQLSTATE[08006] [7] invalid integer value "sslmode=prefer" for connection option "port" (Connection: pgsql, SQL: select * from information_schema.tables where table_catalog = firefly and table_schema = public and table_name = migrations and table_type = 'BASE TABLE')
  at vendor/laravel/framework/src/Illuminate/Database/Connection.php:801
    797▕                     $this->getName(), $query, $this->prepareBindings($bindings), $e
    798▕                 );
    799▕             }
    800▕
  ➜ 801▕             throw new QueryException(
    802▕                 $this->getName(), $query, $this->prepareBindings($bindings), $e
    803▕             );
    804▕         }
    805▕     }
      +35 vendor frames
  36  artisan:35
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
</code></pre>
</details>

In [config/database.php](https://github.com/firefly-iii/firefly-iii/blob/7db21612a03443e86248aec4537fef58aa7294bf/config/database.php), `mysql` has key for `unix_socket`, but `pgsql` does not.

```
        'mysql'  => [
            'driver'      => 'mysql',
            'host'        => envNonEmpty('DB_HOST', $host),
            'port'        => envNonEmpty('DB_PORT', $port),
            'database'    => envNonEmpty('DB_DATABASE', $database),
            'username'    => envNonEmpty('DB_USERNAME', $username),
            'password'    => env('DB_PASSWORD', $password),
            'unix_socket' => env('DB_SOCKET', ''),
            'charset'     => 'utf8mb4',
            'collation'   => 'utf8mb4_unicode_ci',
            'prefix'      => '',
            'strict'      => true,
            'engine'      => 'InnoDB',
            'options'     => $mySqlSSLOptions,
        ],
        'pgsql'  => [
            'driver'      => 'pgsql',
            'host'        => envNonEmpty('DB_HOST', $host),
            'port'        => envNonEmpty('DB_PORT', $port),
            'database'    => envNonEmpty('DB_DATABASE', $database),
            'username'    => envNonEmpty('DB_USERNAME', $username),
            'password'    => env('DB_PASSWORD', $password),
            'charset'     => 'utf8',
            'prefix'      => '',
            'search_path' => envNonEmpty('PGSQL_SCHEMA', 'public'),
            'schema'      => envNonEmpty('PGSQL_SCHEMA', 'public'),
            'sslmode'     => envNonEmpty('PGSQL_SSL_MODE', 'prefer'),
            'sslcert'     => envNonEmpty('PGSQL_SSL_CERT'),
            'sslkey'      => envNonEmpty('PGSQL_SSL_KEY'),
            'sslrootcert' => envNonEmpty('PGSQL_SSL_ROOT_CERT'),
        ],
```

**Comment by JC5** [comment]
You're the first who ever asked, I'm not sure if the framework supports it.

**Reply by charmoniumQ** [reply]
Laravel does support it. This `.env` and `config/database.php` works perfectly in a new Laravel project. However, I can't get it to work in Firefly.

```
# .env
DB_CONNECTION=pgsql
DB_HOST=localhost
DB_DATABASE=sam
DB_USERNAME=sam
DB_SOCKET=/home/sam/Downloads/test2/sockets
```

```
# config/database.php
'pgsql'  => [
            'driver'      => 'pgsql',
            'host'        => envNonEmpty('DB_HOST', $host),
            'port'        => envNonEmpty('DB_PORT', (envNonEmpty('DB_SOCKET', '') === '') ? $port : null),
            'database'    => envNonEmpty('DB_DATABASE', $database),
            'username'    => envNonEmpty('DB_USERNAME', $username),
            'password'    => envNonEmpty('DB_PASSWORD', $password),
            'charset'     => 'utf8',
            'prefix'      => '',
            'search_path' => envNonEmpty('PGSQL_SCHEMA', 'public'),
            'schema'      => envNonEmpty('PGSQL_SCHEMA', 'public'),
            'sslmode'     => envNonEmpty('DB_SSL_MODE', (envNonEmpty('DB_SOCKET', '') === '') ? 'prefer' : null),
            'sslcert'     => envNonEmpty('PGSQL_SSL_CERT'),
            'sslkey'      => envNonEmpty('PGSQL_SSL_KEY'),
            'sslrootcert' => envNonEmpty('PGSQL_SSL_ROOT_CERT'),
            'unix_socket' => envNonEmpty('DB_SOCKET', ''),
        ],
```

When I try this in Firefly, Firefly is still trying to use a port!

```
Mar 26 10:55:01 home-server firefly-iii-setup-start[1621853]:    Illuminate\Database\QueryException
Mar 26 10:55:01 home-server firefly-iii-setup-start[1621853]:   SQLSTATE[08006] [7] connection to server at "localhost" (127.0.0.1), port 5432 failed: fe_sendauth: no password supplied (Connection: pgsql, SQL: select * from information_schema.tables where table_catalog = firefly and table_schema = public and table_name = migrations and table_type = 'BASE TABLE')
```

Is there somewhere in code the port could be set from besides the `$config` object?

**Reply by JC5** [reply]
I see a different solution here:

https://github.com/laravel/framework/issues/3987

Leave DB_HOST empty, should work.

**Reply by stackcoder** [reply]
When checking [databases.php](https://github.com/firefly-iii/firefly-iii/blob/main/config/database.php), the option `unix_socket` is missing in case of pgsql.
